src/main/java/com/mixpanel/android/mpmetrics/MixpanelPushNotification.java

```diff
@@ -290,31 +289,193 @@ protected void maybeAddActionButtons() {
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT_WATCH) {
             for (int i = 0; i < data.buttons.size(); i++) {
                 NotificationButtonData btn = data.buttons.get(i);
-                builder.addAction(this.createAction(btn.icon, btn.label, btn.uri));
+                builder.addAction(this.createAction(btn.icon, btn.label, btn.onTap, btn.buttonId, i + 1));
             }
         }
     }
 
-    @TargetApi(20)
-    protected Notification.Action createAction(int icon, CharSequence title, String uri) {
-        return (new Notification.Action.Builder(icon, title, createActionIntent(uri))).build();
+    protected List<NotificationButtonData> buildButtons(String buttonsJsonStr) {
+        List<NotificationButtonData> buttons = new ArrayList<>();
+        if (null != buttonsJsonStr) {
+            try {
+                JSONArray buttonsArr = new JSONArray(buttonsJsonStr);
+                for (int i = 0; i < buttonsArr.length(); i++) {
+                    JSONObject buttonObj = buttonsArr.getJSONObject(i);
+
+                    // get button icon from name if one sent
+                    int btnIcon = NotificationData.NOT_SET;
+                    if (buttonObj.has("icnm")) {
+                        String btnIconName = buttonObj.getString("icnm");
+                        if (drawableIds.knownIdName(btnIconName)) {
+                            btnIcon = drawableIds.idFromName(btnIconName);
+                        }
+                    }
+
+                    // handle button label
+                    final String btnLabel = buttonObj.getString("lbl");
+
+                    // handle button action
+                    final JSONObject pushActionJSON = buttonObj.getJSONObject("ontap");
+                    final PushTapTarget target = PushTapTarget.fromString(pushActionJSON.getString("type"));
+                    final PushTapAction pushAction = new PushTapAction(target, pushActionJSON.getString("uri"));
```

Comment:
`uri` can be null in the case of "homescreen" type right? Also, how do we pass in whether to open uri in `webview` or `browser` in the button case? Shouldn't we just be passing `ontap` to `buildOnTap(buttonObj.getString("ontap"))`? Let's add a test case for this if I'm correct.